packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/FLTCam.swift

```diff
@@ -0,0 +1,1547 @@
+
+  private var videoWriter: FLTAssetWriter?
+  private var videoWriterInput: FLTAssetWriterInput?
+  private var audioWriterInput: FLTAssetWriterInput?
+  private var assetWriterPixelBufferAdaptor: FLTAssetWriterInputPixelBufferAdaptor?
+  private var videoAdaptor: FLTAssetWriterInputPixelBufferAdaptor?
+
+  /// A dictionary to retain all in-progress FLTSavePhotoDelegates. The key of the dictionary is the
+  /// AVCapturePhotoSettings's uniqueID for each photo capture operation, and the value is the
+  /// FLTSavePhotoDelegate that handles the result of each photo capture operation. Note that photo
+  /// capture operations may overlap, so FLTCam has to keep track of multiple delegates in progress,
+  /// instead of just a single delegate reference.
+  private(set) var inProgressSavePhotoDelegates = [Int64: FLTSavePhotoDelegate]()
+
+  private var imageStreamHandler: FLTImageStreamHandler?
+
+  private var textureId: Int64?
+  private var previewSize: CGSize?
+  private var deviceOrientation = UIDeviceOrientation.unknown
+
+  /// Tracks the latest pixel buffer sent from AVFoundation's sample buffer delegate callback.
+  /// Used to deliver the latest pixel buffer to the flutter engine via the `copyPixelBuffer` API.
+  private var latestPixelBuffer: CVPixelBuffer?
+
+  private var videoRecordingPath: String?
+  private var isRecording = false
+  private var isRecordingPaused = false
+  private var isFirstVideoSample = false
+  private var videoIsDisconnected = false
+  private var audioIsDisconnected = false
+  private var isAudioSetup = false
+  private var lastVideoSampleTime = CMTime.zero
+  private var lastAudioSampleTime = CMTime.zero
+  private var videoTimeOffset = CMTime.zero
+  private var audioTimeOffset = CMTime.zero
+
+  /// True when images from the camera are being streamed.
+  private(set) var isStreamingImages = false
+
+  /// Number of frames currently pending processing.
+  private var streamingPendingFramesCount = 0
+
+  /// Maximum number of frames pending processing.
+  /// To limit memory consumption, limit the number of frames pending processing.
+  /// After some testing, 4 was determined to be the best maximuńm value.
+  /// https://github.com/flutter/plugins/pull/4520#discussion_r766335637
+  private var maxStreamingPendingFramesCount = 4
+
+  private var fileFormat = FCPPlatformImageFileFormat.jpeg
+  private var lockedCaptureOrientation = UIDeviceOrientation.unknown
+  private var exposureMode = FCPPlatformExposureMode.auto
+  private var focusMode = FCPPlatformFocusMode.auto
+  private var flashMode: FCPPlatformFlashMode
+
+  private static func flutterErrorFromNSError(_ error: NSError) -> FlutterError {
+    return FlutterError(
+      code: "Error \(error.code)",
+      message: error.localizedDescription,
+      details: error.domain)
+  }
+
+  // Returns frame rate supported by format closest to targetFrameRate.
+  private static func bestFrameRate(for format: FLTCaptureDeviceFormat, targetFrameRate: Double)
```

Comment:
are these 2 functions stateless? maybe moving them out to utility helpers since this class is getting large.

Reply:
could you also look at other functions and try to move the stateless ones out?